file/org.wso2.carbon.transport.file/src/main/java/org/wso2/carbon/transport/file/connector/server/FileConsumer.java

```diff
@@ -197,6 +205,16 @@ private void setupParams() throws ServerConnectorException {
                         timeOutInterval + " milliseconds", e);
             }
         }
+        String strDeleteIfNotAck = fileProperties.get(Constants.FILE_DELETE_IF_NOT_ACKNOWLEDGED);
+        if (strDeleteIfNotAck != null) {
+            try {
+                deleteIfNotAck = Boolean.parseBoolean(strDeleteIfNotAck);
+            } catch (RuntimeException re) {
```

Comment:
Removed it in https://github.com/wso2/carbon-transports/pull/228/commits/bd704ce1531f38c52ffc2b20070992a82323887d. I thought `Boolean.parseBoolean()` would throw a `RuntimeException` if the argument given is neither `true` nor `false`. I just looked at the implementation and it returns `false` for any argument other than `true` :(